### ISSUE TITLE
[Bug][A11Y] Chip: wrong accessible name and missing role

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-flutter#375]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

iPhone 14+

### OS Version

iOS 18.5

### App version

0.5.0

### Describe the issue

**Wrong accessible name:**
- Chip with text+icon (or in selected state) is read : **"Label, image Label"**
- Chip with icon only is read **"Image"**

**No role button** 

### Expected Behavior

- When the chip has a visible text, the icon should be hidden to accessibility tree, in order to hear only the text as the chip's label
- When the chip has only an image, the image should have an accessible name
- As the chip can be tapped, it should have a "button" role

**Comment by pya35** [comment]
linked to #212 and #213

**Comment by nouha06** [comment]
@pya35 for this point : 
- When the chip has only an image, the image should have an accessible name
waht should be the accessible name in this case ?

**Comment by pya35** [comment]
> [@pya35](https://github.com/pya35) for this point :
> 
>     * When the chip has only an image, the image should have an accessible name
>       waht should be the accessible name in this case ?

@nouha06  The name will be very linked to the context of use. Here for the demo example, the icon is a heart so label could be "favorite" ?

**Comment by nouha06** [comment]
> 
> [@nouha06](https://github.com/nouha06) The name will be very linked to the context of use. Here for the demo example, the icon is a heart so label could be "favorite" ?

But we handle accessibility in the core library, not in the demo app

**Comment by pya35** [comment]
> > [@nouha06](https://github.com/nouha06) The name will be very linked to the context of use. Here for the demo example, the icon is a heart so label could be "favorite" ?
> 
> But we handle accessibility in the core library, not in the demo app

@nouha06 So, maybe you should check with @MaximeTonnerre to know the purpose of this icon

**Comment by pya35** [comment]
@nouha06 and what happens if a project decide to use another icon that the heart one ?

**Comment by nouha06** [comment]
> [@nouha06](https://github.com/nouha06) and what happens if a project decide to use another icon that the heart one ?

we can change the icon, but the accessible name of the image should be generic in this case

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (406)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 406
cc @B3nz01d (product owner)
cc @Tayebsed93 @nouha06 @AhmedAmineZr (dev team)
cc @jerome-regnier (design team)

**Comment by nouha06** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (alpha) (406)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 406
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)

**Comment by pya35** [comment]
@nouha06 Suggestion Chip is now read as "**_Chip labelLabel, Button_**", seems to have a double label here (no problem with filter chip label) 
Observed on iOS, not tested on Android

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (425)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 375
cc @B3nz01d (product owner)
cc @Tayebsed93 @nouha06 @AhmedAmineZr (dev team)
cc @jerome-regnier (design team)

**Comment by nouha06** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (alpha) (425)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 375
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)

**Comment by B3nz01d** [comment]
@nouha06 
In this version, the Suggestion chip has a bug in the display and it cannot be tested anymore.

The test on the Filter chip is ok for me.

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (429)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 375
cc @B3nz01d (product owner)
cc @Tayebsed93 @nouha06 @AhmedAmineZr (dev team)
cc @jerome-regnier (design team)

**Comment by nouha06** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (alpha) (429)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 375
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)

**Comment by B3nz01d** [comment]
@nouha06  when the voice over or talkback provides the chip's info, it should read the label of the chip, today if I change the label the spoken information is still "chip label, button" but it should read le label itself. 
@pya35  if I change the label to "mobile" what should the vocalization say:
* "Mobile, Button"
* "Suggestion chip: Mobile, Button"
How does the user understand the concept of the suggestion chip?
The same question applye for the Filter chip

**Comment by pya35** [comment]
@nouha06 @B3nz01d yes, the **visible label** should be read by screen readers, so if we edit the text like "mobile" we should hear "mobile, button"

Regarding the filter or suggestion concept, this is very dependant to context of use and the demo app is not appropriated for that as components are not presented in a representative demo use-case. 

From a11y POV, the accessible name shall start by the visible text (for example "mobile"), so if we set "filter by mobile" this will not work.
This kind of components should be grouped and the group should be labelled in a way that we understand the concept of filter or suggestion. (for example "filter by type of device").

I also suggest to add an optional hint to the button, to add some more information if needed in complex cases.

This proposal should be applied to every apps.

@pylapp @ludovic35 @paulinea @florentmaitre @MaximeTonnerre

**Comment by B3nz01d** [comment]
@pya35 have you stated on the right output in this case?

**Comment by pya35** [comment]
@B3nz01d @nouha06 
Chips shall be grouped together:
- the group shall be named and configurable (contextual), for example : "Filter by device brand" 
- each chip shall contain a configurable text, for example "Apple", "Samsung"

**Comment by Tayebsed93** [comment]
Hello @pya35.
The accessibility requirement is well understood.
For now, the chip will remain independent.
A discussion will take place between @paulinea and @B3nz01d  to evaluate the relevance of creating a grouped chips component.
This topic will not be addressed in the scope of this ticket; if needed, a new ticket will be created to implement this behavior.

**Comment by Tayebsed93** [comment]
@pya35 Aside from the grouped chip issue, is the rest of the ticket good for you?
If so, you can add the “OK Accessibility” label.

**Comment by pya35** [comment]
OK for accessible name and role

**Comment by B3nz01d** [comment]
Closed during Sprint # 23
